### ISSUE TITLE
Failed to mount NFS file system on RHEL 7.2

### ISSUE THREAD

**Issue by lcorreia** [kimchi-project/ginger#428]
GINFS00018E: NFS mount failed, 'str' object has no attribute 'iteritems'

**Comment by biancafc** [comment]
Getting the same issue in Fedora 25.

**Comment by danielhb** [comment]
Is this an Invalid Parameter or an Operation Failed error?

```
[danielhb@arthas ginger]$ grep -R 'GINFS00018E' .
./i18n.py:    "GINFS00018E": _("NFS mount failed, %(err)s"),
Binary file ./model/fs_utils.pyc matches
./model/filesystem.py:                raise InvalidParameter("GINFS00018E", {"err": e})
./model/fs_utils.py:        raise OperationFailed("GINFS00018E", err)
Binary file ./model/filesystem.pyc matches
Binary file ./i18n.pyc matches
[danielhb@arthas ginger]$ 
```

**Comment by biancafc** [comment]
My wok error log:

```
rc: 32 error: mount.nfs: access denied by server while mounting 192.168.157.137:/home/bianca/Documentos/kimchi-release-tests/wok/src/wok/plugins/ginger/test
 returned from cmd: mount 192.168.157.137:/home/bianca/Documentos/kimchi-release-tests/wok/src/wok/plugins/ginger/test /home/bianca/Documentos/kimchi-release-tests/wok/src/wok/plugins/ginger/test
GINFS00018E: NFS mount failed, 'str' object has no attribute 'iteritems'
```

**Comment by danielhb** [comment]
@biancafc  The patch '[Ginger] Bug fix #428: wrong parameter in OperationFailed message' in the mailing list fixes this behavior by forwarding the actual error message that happened in the mount command.

I've created an environment that reproduces the same error message you're supposed to have by your log. Here's how the UI looks after applying the patch:

![image](https://cloud.githubusercontent.com/assets/866780/20940716/e9065d3e-bbda-11e6-9a8b-326879a8133f.png)


Here is the WoK log:

```
rc: 32 error: mount.nfs: access denied by server while mounting 127.0.0.1:/mnt/nfs
 returned from cmd: mount 127.0.0.1:/mnt/nfs /mnt/nfs
ERROR:cherrypy.error:rc: 32 error: mount.nfs: access denied by server while mounting 127.0.0.1:/mnt/nfs
 returned from cmd: mount 127.0.0.1:/mnt/nfs /mnt/nfs
GINFS00018E: NFS mount failed, mount.nfs: access denied by server while mounting 127.0.0.1:/mnt/nfs

```

Note that the "  'str' object has no attribute 'iteritems' " error is gone, resulting only in the actual error message. Also note that the actual error message is most likely something wrong with your setup, not necessarily a problem in the Ginger NFS backend. I suggest double checking the settings in the /etc/exports file.

**Comment by biancafc** [comment]
I've tested again and could not reproduce this issue anymore. Working fine with Fedora 25.
